src/main/java/com/aws/greengrass/config/ConfigurationReader.java

```diff
@@ -97,9 +115,46 @@ public static void mergeTLogInto(Configuration config, Path tlogPath, boolean fo
         }
     }
 
-    private static void mergeTLogInto(Configuration c, Path p) throws IOException {
+    private static void mergeTLogInto(Configuration c, Path p, ConfigurationMode configurationMode) throws IOException {
         try (BufferedReader bufferedReader = Files.newBufferedReader(p)) {
-            mergeTLogInto(c, bufferedReader, false, null);
+            mergeTLogInto(c, bufferedReader, false, null, configurationMode);
+        }
+    }
+
+    /**
+     * In place update for the given config object from the given transaction log, adhering to the given update behavior
+     * tree and without losing listeners.
+     *
+     * @param config         configuration to merge into
+     * @param tlogPath       path of the tlog file to read to-be-replaced config from
+     * @param forceTimestamp should ignore if the proposed timestamp is older than current
+     * @param mergeCondition Predicate that returns true if the provided Topic should be merged and false if not
+     * @param tree           Merge behavior hierarchy for the update
+     * @throws IOException if update fails
+     */
+    public static void updateFromTLog(Configuration config, Path tlogPath, boolean forceTimestamp,
```

Comment:
Deployment task thread

Reply:
I think we need to run it on the context thread so that we don't have partial config changes start firing subscriptions.

Reply:
Rollbacks never had it, but I agree. I've changed rollback config change to run on the publish thread

Reply:
If there are specific threads that should or should not call this, can you add that to the javadoc?

Reply:
It will depend on what you use this for. For instance if you were to use this method on nucleus startup before any services have been started up, then there's no need to synchronize. But I added some notes to the javadoc